contracts/liquidity-pool/src/contract.rs

```
@@ -432,6 +434,31 @@ fn query_reserve<S: Storage, A: Api, Q: Querier>(
     Ok(ReserveResponse { ma_token_address })
 }
 
+fn query_reserves_list<S: Storage, A: Api, Q: Querier>(
+    deps: &Extern<S, A, Q>,
+) -> StdResult<ReservesListResponse> {
+    let reserves = reserves_state_read(&deps.storage);
+
+    let reserves_list: StdResult<Vec<_>> = reserves
+        .range(None, None, Order::Ascending)
+        .map(|item| {
+            let (k, v) = item?;
+            let denom = String::from_utf8(k).unwrap();
+            let ma_token_address = deps
+                .api
+                .human_address(&CanonicalAddr::from(v.ma_token_address))?;
+            Ok(ReserveInfo {
+                denom,
+                ma_token_address,
+            })
+        })
+        .collect();
+
+    Ok(ReservesListResponse {
+        reserves_list: reserves_list?,
```

If the result of collect returns an error maybe we should return an error too? (Just calling ? after the `collect` method and having reserves_list be a Vec instead of a StdResult)